pkg/cmd/grafana-cli/commands/scuemata_validation_command_test.go

```diff
@@ -0,0 +1,102 @@
+package commands
+
+import (
+	"embed"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/grafana/grafana/pkg/schema/load"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+//go:embed testdata/public/*/*.cue testdata/public/*/plugin.json
+var base embed.FS
+var pluginSchema, _ = fs.Sub(base, "testdata/public")
+
+func TestValidateScuemataBasics(t *testing.T) {
+	t.Run("Testing scuemata validity with valid cue schemas", func(t *testing.T) {
+		tempDir := os.DirFS(filepath.Join("testdata", "valid_scuemata"))
+
+		var baseLoadPaths = load.BaseLoadPaths{
+			BaseCueFS:       tempDir,
+			DistPluginCueFS: pluginSchema,
+		}
+
+		err := validate(nil, baseLoadPaths, load.BaseDashboardFamily)
+		require.NoError(t, err, "error while loading base dashboard scuemata")
+
+		err = validate(nil, baseLoadPaths, load.DistDashboardFamily)
+		require.NoError(t, err, "error while loading dist dashboard scuemata")
+	})
+
+	t.Run("Testing scuemata validity with invalid cue schemas - family missing", func(t *testing.T) {
+		tempDir := os.DirFS(filepath.Join("testdata", "invalid_scuemata_missing_family"))
+
+		var baseLoadPaths = load.BaseLoadPaths{
+			BaseCueFS:       tempDir,
+			DistPluginCueFS: pluginSchema,
+		}
+
+		err := validate(nil, baseLoadPaths, load.BaseDashboardFamily)
+		assert.EqualError(t, err, "error while loading dashboard scuemata, err: dashboard schema family did not exist at expected path in expected file")
+	})
+
+	t.Run("Testing scuemata validity with invalid cue schemas - panel missing", func(t *testing.T) {
+		tempDir := os.DirFS(filepath.Join("testdata", "invalid_scuemata_missing_panel"))
+
+		var baseLoadPaths = load.BaseLoadPaths{
+			BaseCueFS:       tempDir,
+			DistPluginCueFS: pluginSchema,
+		}
+
+		err := validate(nil, baseLoadPaths, load.BaseDashboardFamily)
+		require.NoError(t, err, "error while loading base dashboard scuemata")
+
+		err = validate(nil, baseLoadPaths, load.DistDashboardFamily)
+		assert.EqualError(t, err, "all schema should be valid with respect to basic CUE rules, Family.lineages.0.0: field #Panel not allowed")
+	})
+
+	t.Run("Testing scuemata validity against JSON files", func(t *testing.T) {
```

Comment:
I started a framework for testing dashboard JSON [here](https://github.com/grafana/grafana/blob/1b8c0ce88bdc3cc22564edb535b81863799b68c5/pkg/schema/load/load_test.go#L50) (idk why i didn't point you to this earlier, sorry!). This subtest appears to duplicate that, albeit with both valids and invalids. So, two comments there:

1. Let's keep this PR only to checking the validity of scuemata themselves, in keeping with the other comment about having a separate command for checking resources
2. While i'm not wedded to having the test i wrote live where it does today, there almost certainly should be only one framework for validating test input JSON against schema

(also, nit: The title of this test seems to flow in the wrong direction. Scuemata/schema can determine whether JSON/resources are valid, not the other way around.)

Reply:
> Scuemata/schema can determine whether JSON/resources are valid, not the other way around.

Though, i should maybe clarify: _right now_, the JSON generated by the frontend application does not pass our schemas. That's a temporary situation that we absolutely have to fix in the coming weeks. Once we've fixed it, it should never occur again (barring bugs).